lib/KRadioButtonGroup.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+
+  <div class="k-radio-button-group">
+    <!-- @slot To render the KRadioButtons -->
+    <slot></slot>
+  </div>
+
+</template>
+
+
+<script>
+
+  export default {
+    name: 'KRadioButtonGroup',
+    props: {
+      /**
+       * Specifies whether the radio button group is enabled or disabled.
+       * Used to make the first radio button active when the radio group becomes enabled again after being disabled.
+       */
+      enable: {
+        type: Boolean,
+        default: true,
+      },
+    },
+    data() {
+      return {
+        radioButtons: [],
+        focusedRadioIdx: 0,
+        firstRadioIdx: 0,
+        lastRadioIdx: 0,
+      };
+    },
+    computed: {
+      isFirefox() {
+        return navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
+      },
+    },
+    watch: {
+      // To focus on the first radio button when radio group enables again
+      enable(newVal, oldVal) {
+        if (newVal && !oldVal && this.radioButtons.length > 0) {
+          this.setChecked(0);
+          this.focusedRadioIdx = 0;
+        }
+      },
+    },
+    mounted() {
+      this.$nextTick(() => {
+        if (this.isFirefox) {
+          if (this.$children[0].$options._componentTag === 'KGridItem') {
+            this.$children.forEach(gridItem => {
+              gridItem.$children.forEach(fixedGridItem => {
+                fixedGridItem.$children.forEach(radioBtn => {
+                  this.radioButtons.push(radioBtn);
+                  radioBtn.$el.addEventListener('keyup', this.onKeyUp);
```

Comment:
> I would also recommend doing the very same thing for the `input` event (line 71)

For the `keyup` listener I am able to have the listener only on the parent `KRadioButtonGroup` But I have issues with putting `input` listener on the parent.

So to query all the `KRadioButton` and put all those in a list I have used recursion as you have told me.
```vue
// const KRadioButtonGroupChildren = this.$children;
// this.queryAndAddRadioBtns(KRadioButtonGroupChildren);

 queryAndAddRadioBtns(children) {
        if (children[0].$options._componentTag === 'KRadioButton') {
          for (let i = 0; i < children.length; i++) {
            const radioBtnVueComp = children[i];
            this.radioButtons.push(radioBtnVueComp);
            radioBtnVueComp.setTabIndex(-1);
          }
          return;
        }
        this.queryAndAddRadioBtns(children[0].$children);
        if (children.length > 1) this.queryAndAddRadioBtns(children[1].$children);
        return;
}
```

Now the issue is, 

1. I cannot listen for `input` on `KRadioButtonGroup` component as `input` is emitted by `KRadioButton`. 
 ```vue
 this.$on('input', () => {}); // no no won't work
 ```
I tried to find some answers if some way we can listen for the child event on the parent but did not find anything useful.
If you know how we could listen for the `input` event on parents where the children are rendered with the help of a slot. If you know it would be a great help.

2. The second thing I tried is having an `input` listener and `click` listener on the `KRadioButtonGroup`
and during `queryAndAddRadioBtns` we set a custom `btnIndex` attribute to each input element to know each radio btn index for later purposes.

But when we click on `radio icon` I thought we would get click on `input` element but **NO** we get an event on 
`radio icon svg` element or `div containing the label` (if clicks on the label). Because of this my logic failed because I checked 'k-radio-button-input` className on  `onKeyUp` method
```vue
onKeyUp(event) {
        if (event.target.className === 'k-radio-button-input') {
          const handlers = {
            ArrowLeft: this.focusPreviousRadio,
            ArrowRight: this.focusNextRadio,
            ArrowUp: this.focusPreviousRadio,
            ArrowDown: this.focusNextRadio,
          };
          const handler = handlers[event.key];
          if (handler) {
            event.preventDefault();
            event.stopPropagation();
            handler(event);
          }
        }
}
```
So could you help me here what I can do? currently what I have moved on with is having `keyup` event only on `KRadioButtonGroup` and `on input` on every `KRadioButton`.

Reply:
I see, thank you. 

The main part of my feedback was concerned with `keyup` and I only added `input` because I assumed it would work as well and it'd be clearer to do all events in the same way. I looked into the latter briefly now and I think the issue with that is that it's rather Vue specific event and it doesn't seem to propagate. I had no success with `change` event either.

So here I think we could stay with the original approach?

```
radioBtn.$on('input', () => {
    this.handleInputChange(index);
  });
```

(but while still making sure there's only one `keyup` event listener which is the main issue)

 I'm sorry this was such a hassle, I didn't expect that `input`/`change` will be troubling.

Reply:
I think this should still be possible to do with the recursion approach

Reply:
As far as I can say, I think we can do `input` handler assignment during the recursion,  around time when you're saving a radio button to the list, right?

Reply:
> As far as I can say, I think we can do `input` handler assignment during the recursion, around time when you're saving a radio button to the list, right?
Thank you!! 

Yes, we can do that in the recursion method too.

Reply:
Okay, good!